sentry/src/main/java/io/sentry/Hub.java

```diff
@@ -601,4 +601,57 @@ public void flush(long timeoutMillis) {
     }
     return clone;
   }
+
+  @Override
+  public SentryId captureTransaction(Transaction transaction, Object hint) {
+    SentryId sentryId = SentryId.EMPTY_ID;
+    if (!isEnabled()) {
+      options
+          .getLogger()
+          .log(
+              SentryLevel.WARNING,
+              "Instance is disabled and this 'captureTransaction' call is a no-op.");
+    } else if (transaction == null) {
+      options
+          .getLogger()
+          .log(SentryLevel.WARNING, "captureTransaction called with null parameter.");
+    } else {
+      try {
+        final StackItem item = stack.peek();
+        if (item != null) {
+          sentryId = item.client.captureTransaction(transaction, item.scope, hint);
+        } else {
+          options.getLogger().log(SentryLevel.FATAL, "Stack peek was null when captureTransaction");
+        }
+      } catch (Exception e) {
+        options
+            .getLogger()
+            .log(
+                SentryLevel.ERROR,
+                "Error while capturing event with id: " + transaction.getEventId(),
+                e);
+      }
+    }
+    this.lastEventId = sentryId;
+    return sentryId;
+  }
+
+  @Override
+  public @Nullable Transaction startTransaction(TransactionContexts transactionContexts) {
```

Comment:
What should happen when this method is called but another transaction is already started?

Reply:
at least for sessions, we stop and capture the current one and start a new one, I believe it should be the same thing.